# Description: revert "single click to browser selected items" behaviour
# which was introduced in 3.18, it's confusing to users and was added without
# rational. The issue has been reported upstream but didn't get traction
#
# Ubuntu: https://launchpad.net/bugs/1519778
# Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=758649
Index: gtk+3.0-3.18.7/gtk/gtkfilechooserwidget.c
===================================================================
--- gtk+3.0-3.18.7.orig/gtk/gtkfilechooserwidget.c
+++ gtk+3.0-3.18.7/gtk/gtkfilechooserwidget.c
@@ -2401,7 +2401,10 @@ list_button_press_event_cb (GtkWidget
                    gdk_device_get_source (device) == GDK_SOURCE_TOUCHSCREEN;
 
   get_selection_modifiers (widget, event, &modify, &extend);
-  if (!is_touchscreen &&
+  /* Ubuntu, disable the new behaviour of opening sometime on single click
+     it's confusing for users and the change had no rational
+     https://launchpad.net/bugs/1519778 */
+  if (FALSE && !is_touchscreen &&
       !modify && !extend &&
       event->type == GDK_BUTTON_PRESS &&
       event->button == GDK_BUTTON_PRIMARY &&
